app-modules/etno/src/Support/CitationFormatter.php

```
@@ -0,0 +1,107 @@
+        }
+
+        if ($timePeriod) {
+            $parts[] = '('.__('etno::ui.citations.time_period').": {$timePeriod}).";
+        }
+
+        if ($doi) {
+            $parts[] = "https://doi.org/{$doi}";
+        }
+
+        return implode(' ', $parts);
+    }
```

PHPMD (2.15.0)

[warning] The method format() has a Cyclomatic Complexity of 23. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)